timelock-impl/src/main/java/com/palantir/atlasdb/timelock/AsyncTimelockService.java

```diff
@@ -73,6 +76,11 @@ ListenableFuture<GetCommitTimestampsResponse> getCommitTimestamps(
 
     ListenableFuture<TimestampRange> getFreshTimestampsAsync(int timestampsToRequest);
 
+    ListenableFuture<NamedMinTimestampLeaseResponse> acquireNamedMinTimestampLease(
+            TimestampName timestampName, UUID requestId, int numFreshTimestamps);
+
+    ListenableFuture<Long> getMinLeasedNamedTimestamp(TimestampName timestampName);
```

Comment:
Much simpler to expose the smallest unit of work here (as opposed to batch)

Reply:
You will definitely have some fun futures code to write here, to actually implement the batched piece